joomla/joomla-cms
# [4.0] Modulefilterfix

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/30302 .

### Summary of Changes

This pull request turns the links `site modules` and `admin modules` into one link `modules`. So the URI stays the same when changing from site module to admin module. This has the advantage that the filters, especially reset filters, continue to work correctly.

> Note If the distinction  between admin and site modules is wanted, it would be better to include the `client_id `in a hidden form field. But I would do this in a separate PR.

### Testing Instructions

1. Apply this patch. You can use patch tester. `npm` or `composer` are **not** required.

2. Open the modules and check that admin and site modules view can be changed with the drop down and that the modules entry is always active in the admin menu on the left side.

![Modules Site test Administration](https://user-images.githubusercontent.com/9974686/92306206-a374f600-ef8d-11ea-9580-368c983cf95f.png)

3. Set filters, switch between admin and custom and check that the filters are deleted when you switch. See https://github.com/joomla/joomla-cms/issues/30302
4. Check, that quicktask near the content menu item works correct.

![Content Dashboard test Administration](https://user-images.githubusercontent.com/9974686/92306208-a4a62300-ef8d-11ea-8b15-5d022d03487a.png)

5. Use different presets and check, that these work correct.

![Modules Administrator Menu test Administration](https://user-images.githubusercontent.com/9974686/92306209-a53eb980-ef8d-11ea-97a0-f78697274825.png)


### Actual result BEFORE applying this Pull Request

Reset Filter does not work properly, see https://github.com/joomla/joomla-cms/issues/30302.


### Expected result AFTER applying this Pull Request

Reset Filter does work.

### Documentation Changes Required

No
